File: 1221.c
int balancedStringSplit(char* s) {
    int num = 0;
    int sum = 0;
    for (int i = 0; s[i] != '\0'; i++) {
        if (s[i] == 'L') num++;
        else num--;
        if (num == 0) sum++;
    }
    return sum;
}

File: 1221.py
class Solution:
    def balancedStringSplit(self, s: str) -> int:
        balanced = 0
        total = 0
        for i in s:
            if i == 'L':
                balanced -= 1
            else:
                balanced += 1
            if balanced == 0:
                total += 1
        return total
